Sources/Basics/FileSystem+Extensions.swift

```diff
@@ -91,14 +91,30 @@ extension FileSystem {
 }
 
 extension FileSystem {
-    public func getOrCreateSwiftPMConfigurationDirectory(warningHandler: (String) -> Void) throws -> AbsolutePath {
+    public func getOrCreateSwiftPMConfigurationDirectory(warningHandler: @escaping (String) -> Void) throws -> AbsolutePath {
         let idiomaticConfigurationDirectory = self.swiftPMConfigurationDirectory
 
         // temporary 5.6, remove on next version: transition from previous configuration location
         if !self.exists(idiomaticConfigurationDirectory) {
             try self.createDirectory(idiomaticConfigurationDirectory, recursive: true)
         }
 
+        let handleExistingFiles = { (configurationFiles: [AbsolutePath]) in
+            for file in configurationFiles {
+                let destination = idiomaticConfigurationDirectory.appending(component: file.basename)
+                if !self.exists(destination) {
+                    try self.copy(from: file, to: destination)
+                } else {
+                    // Only emit a warning if source and destination file differ in their contents.
+                    let srcContents = try? self.readFileContents(file)
+                    let dstContents = try? self.readFileContents(destination)
```

Comment:
should we let these throw?

Reply:
I think it's fine this way, matches what we're doing in the initial scan